src/domains/ecommerce-app/api/ecommerce-user-stats-service/v1/_openapi.json.tpl

```diff
@@ -0,0 +1,228 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "version": "0.0.1",
+    "title": "Pagopa eCommerce user stats service",
+    "description": "This microservice handles statistics for payment performed by an authenticated used on eCommerce",
+    "contact": {
+      "name": "pagoPA - Touchpoints team"
+    }
+  },
+  "tags": [
+    {
+      "name": "user-stats",
+      "description": "Api's for tracing and retrieve user statistics",
+      "externalDocs": {
+        "url": "TODO",
+        "description": "Technical specifications"
+      }
+    }
+  ],
+  "externalDocs": {
+    "url": "https://pagopa.atlassian.net/wiki/spaces/I/pages/492339720/pagoPA+eCommerce+Design+Review",
+    "description": "Design review"
+  },
+  "servers": [
+    {
+      "url": "https://${hostname}"
+    }
+  ],
+  "security": [
+    {
+      "ApiKeyAuth": []
+    }
+  ],
+  "paths": {
+    "/user/lastPaymentMethodUsed": {
+      "post": {
+        "parameters": [
+          {
+            "in": "header",
+            "name": "x-user-id",
```

Comment:
curiosity, why in the header and not in the body?

Reply:
in the `header` it is useful when the data is used at apim level, in this case is there an authorization mechanism that identifies that api can be called for that payment flow? so for a given transaction?

Reply:
This POST api call will be called by event-dispatcher in a b2b interaction, the only identified auth mechanism here is key.

The GET api call, instead, will be called by IO app (and so authenticated with session token, so user id will be injected to the service as header parameter).

I put this information here to have _homogeneity_ between the POST and GET api calls, but it can be moved inside POST body, the modification is pretty simple here.

What do you think?